docs/Developer-guide.md

```diff
@@ -0,0 +1,131 @@
+# Developer guide
```

Comment:
Meta: suggest linking to this document from the README under "contribution guidelines".

Reply:
Let me take some time to formalize a better home page, added as an issue #15 for first milestone